module/buffer.go

```diff
@@ -3,19 +3,18 @@ package module
 import (
 	"github.com/onflow/flow-go/model/cluster"
 	"github.com/onflow/flow-go/model/flow"
-	"github.com/onflow/flow-go/model/messages"
 )
 
 // PendingBlockBuffer defines an interface for a cache of pending blocks that
 // cannot yet be processed because they do not connect to the rest of the chain
 // state. They are indexed by parent ID to enable processing all of a parent's
 // children once the parent is received.
 type PendingBlockBuffer interface {
-	Add(originID flow.Identifier, proposal *messages.BlockProposal) bool
+	Add(originID flow.Identifier, block *flow.Block) bool
 
-	ByID(blockID flow.Identifier) (*flow.PendingBlock, bool)
+	ByID(blockID flow.Identifier) (*flow.Slashable[flow.Block], bool)
 
-	ByParentID(parentID flow.Identifier) ([]*flow.PendingBlock, bool)
+	ByParentID(parentID flow.Identifier) ([]*flow.Slashable[flow.Block], bool)
```

Comment:
I would strongly prefer to use `flow.Slashable` instead of `*flow.Slashable` here since it's a structure that just holds two pointers, no need to allocate it on heap

Reply:
Agree - good point